packages/redux-simple-auth/src/middleware.js

```diff
@@ -120,6 +126,35 @@ export default (config = {}) => {
             return response
           })
         }
+        case INVALIDATE_SESSION: {
+          const state = getState()
+          const authenticatorName = getAuthenticator(state)
+
+          if (authenticatorName) {
+            const authenticator = findAuthenticator(authenticatorName)
+
+            if (!authenticator) {
+              throw new Error(
+                `No authenticator with name \`${action.meta.authenticator}\` ` +
+                  'was found. Be sure you have defined it in the authenticators ' +
+                  'config.'
+              )
+            }
+
+            if (authenticator) {
+              authenticator.invalidate(getSessionData(state)).then(
+                // dispatch session invalidation to allow things outside this
+                // direct ecosystem to be updated.
+                () => next(action),
+                () => dispatch(invalidateSessionFailed())
+              )
+            } else {
+              dispatch(invalidateSessionFailed())
+            }
+          }
+          // else presumably there was no authenticated session.
+          break
```

Comment:
We should `return` something here. Since this middleware acts as an interceptor, we want to push through the original result of the action.

I actually discovered a bug and had to [fix this just recently]((https://github.com/jerelmiller/redux-simple-auth/pull/32)).

Since we likely want to return the same type here, we should return a resolved promise even in situations where we don't end up with the `invalidate` call on `authenticator`. This would ensure that someone could always `dispatch(invalidateSession()).then`.

Reply:
💯